handlers/download.go

```diff
@@ -62,6 +93,33 @@ func (s Server) entryGet() http.HandlerFunc {
 	}
 }
 
+func renderPassphrasePrompt(w http.ResponseWriter, r *http.Request, id picoshare.EntryID, meta picoshare.UploadMetadata) {
+	renderPassphrasePromptWithError(w, r, id, meta, "")
+}
+
+func renderPassphrasePromptWithError(w http.ResponseWriter, r *http.Request, id picoshare.EntryID, meta picoshare.UploadMetadata, errorMsg string) {
+	// Minimal inline template to avoid adding static assets yet.
+	const tpl = `<!doctype html>
+<html>
+<head><meta charset="utf-8"><title>PicoShare - Protected</title></head>
+<body>
+<h1>Enter passphrase</h1>
+{{if .Error}}<p style="color:red">{{.Error}}</p>{{end}}
+<form method="post" action="/-{{.ID}}">
+  <input type="password" name="passphrase" placeholder="Passphrase" required autofocus />
+  <button type="submit">View</button>
+  {{if .Filename}}<p>File: {{.Filename}}</p>{{end}}
+</form>
+</body>
+</html>`
+	t := template.Must(template.New("pp").Parse(tpl))
+	_ = t.Execute(w, struct {
+		ID       string
+		Filename string
+		Error    string
+	}{ID: id.String(), Filename: meta.Filename.String(), Error: errorMsg})
+}
+
```

Comment:
We should add an e2e test to cover the scenario of uploading a file with a passphrase and then downloading it with a passphrase.